File: CS/RangedMaskSample/MaskButtonEditMaskFormatElement.cs
using System;
using DevExpress.Data.Mask;
using System.Globalization;

namespace RangedMaskSample
{
    class MaskButtonEditMaskFormatElement : DateTimeNumericRangeFormatElementEditable
    {
        private int aIndex;
        private int initvalue;
        private MaskButtonEdit aEdit;

        public MaskButtonEditMaskFormatElement(string mask, DateTimeFormatInfo datetimeFormatInfo, int PartNumber, MaskButtonEdit aedit)
            : base(mask, datetimeFormatInfo, DateTimePart.Time)
        {
            aIndex = PartNumber;
            aEdit = aedit;
        }

        public override DateTimeElementEditor CreateElementEditor(DateTime editedDateTime)
        {
            MaskElement aElement = aEdit.Properties.MaskElements[aIndex];
            EditElement editElement = aElement.EditCollection.Find(aEdit);
            if (editElement != null)
                initvalue = editElement.EditValue;
            else
                initvalue = 0;
            return new DateTimeNumericRangeElementEditor(initvalue, aElement.MinValue, aElement.MaxValue, 1, Math.Max(aElement.MinValue.ToString().Length, aElement.MaxValue.ToString().Length));
        }

        public override DateTime ApplyElement(int result, DateTime editedDateTime)
        {
            MaskElement aElement = aEdit.Properties.MaskElements[aIndex];
            EditElement editElement = aElement.EditCollection.Find(aEdit);
            if (editElement != null)
                editElement.EditValue = result;
            return new DateTime(editedDateTime.Ticks + 1);
        }

        public override string Format(DateTime formattedDateTime)
        {
            MaskElement aElement = aEdit.Properties.MaskElements[aIndex];
            EditElement editElement = aElement.EditCollection.Find(aEdit);
            if (editElement != null)
                return editElement.EditValue.ToString();
            initvalue = 0;
            return initvalue.ToString();
        }
    }
}

File: VB/RangedMaskSample/MaskButtonEditMaskFormatElement.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Data.Mask
Imports System.Globalization

Namespace RangedMaskSample
	Friend Class MaskButtonEditMaskFormatElement
		Inherits DateTimeNumericRangeFormatElementEditable
		Private aIndex As Integer
		Private initvalue As Integer
		Private aEdit As MaskButtonEdit

		Public Sub New(ByVal mask As String, ByVal datetimeFormatInfo As DateTimeFormatInfo, ByVal PartNumber As Integer, ByVal aedit As MaskButtonEdit)
			MyBase.New(mask, datetimeFormatInfo, DateTimePart.Time)
			aIndex = PartNumber
			Me.aEdit = aedit
		End Sub

		Public Overrides Function CreateElementEditor(ByVal editedDateTime As DateTime) As DateTimeElementEditor
			Dim aElement As MaskElement = aEdit.Properties.MaskElements(aIndex)
			Dim editElement As EditElement = aElement.EditCollection.Find(aEdit)
			If editElement IsNot Nothing Then
				initvalue = editElement.EditValue
			Else
				initvalue = 0
			End If
			Return New DateTimeNumericRangeElementEditor(initvalue, aElement.MinValue, aElement.MaxValue, 1, Math.Max(aElement.MinValue.ToString().Length, aElement.MaxValue.ToString().Length))
		End Function

		Public Overrides Function ApplyElement(ByVal result As Integer, ByVal editedDateTime As DateTime) As DateTime
			Dim aElement As MaskElement = aEdit.Properties.MaskElements(aIndex)
			Dim editElement As EditElement = aElement.EditCollection.Find(aEdit)
			If editElement IsNot Nothing Then
				editElement.EditValue = result
			End If
			Return New DateTime(editedDateTime.Ticks + 1)
		End Function

		Public Overrides Function Format(ByVal formattedDateTime As DateTime) As String
			Dim aElement As MaskElement = aEdit.Properties.MaskElements(aIndex)
			Dim editElement As EditElement = aElement.EditCollection.Find(aEdit)
			If editElement IsNot Nothing Then
				Return editElement.EditValue.ToString()
			End If
			initvalue = 0
			Return initvalue.ToString()
		End Function
	End Class
End Namespace
